src/gpu/d3d11/SDL_gpu_d3d11.c

```diff
@@ -3733,28 +3746,432 @@ static SDL_GpuCommandBuffer* D3D11_AcquireCommandBuffer(
+    {
+        D3D11_INTERNAL_CycleActiveBuffer(
+            renderer,
+            d3d11UniformBuffer->bufferContainer
+        );
+
+        d3d11UniformBuffer->offset = 0;
+
+        D3D11_INTERNAL_TrackBuffer(
+            d3d11CommandBuffer,
+            d3d11UniformBuffer->bufferContainer->activeBuffer
+        );
+    }
+
+    drawOffset = d3d11UniformBuffer->offset;
+
+    D3D11_INTERNAL_SetUniformBufferData(
+        renderer,
+        d3d11CommandBuffer,
+        d3d11UniformBuffer->bufferContainer->activeBuffer,
+        d3d11UniformBuffer->offset,
+        data,
+        dataLengthInBytes
+    );
+
+    d3d11UniformBuffer->offset += d3d11UniformBuffer->currentBlockSize;
 
-	commandBuffer = D3D11_INTERNAL_GetInactiveCommandBufferFromPool(renderer);
-	commandBuffer->windowData = NULL;
-	commandBuffer->graphicsPipeline = NULL;
-	commandBuffer->computePipeline = NULL;
-	commandBuffer->vertexUniformBuffer = NULL;
-	commandBuffer->fragmentUniformBuffer = NULL;
-	commandBuffer->computeUniformBuffer = NULL;
-	for (Uint32 i = 0; i < MAX_COLOR_TARGET_BINDINGS; i += 1)
-	{
-		commandBuffer->colorTargetResolveTexture[i] = NULL;
-		commandBuffer->colorTargetResolveSubresourceIndex[i] = 0;
-		commandBuffer->colorTargetMsaaHandle[i] = NULL;
-	}
+    offsetInConstants = drawOffset / 16;
+    blockSizeInConstants = d3d11UniformBuffer->currentBlockSize / 16;
 
-	D3D11_INTERNAL_AcquireFence(commandBuffer);
-	commandBuffer->autoReleaseFence = 1;
+    for (i = 0; i < d3d11UniformBuffer->bindSlotCount; i += 1)
+    {
+        if (d3d11UniformBuffer->bindSlots[i].shaderStageFlag == SDL_GPU_SHADERSTAGE_VERTEX)
+        {
+            /* stupid workaround for god awful D3D11 drivers
+             * see: https://learn.microsoft.com/en-us/windows/win32/api/d3d11_1/nf-d3d11_1-id3d11devicecontext1-vssetconstantbuffers1#calling-vssetconstantbuffers1-with-command-list-emulation
+             */
+            ID3D11DeviceContext1_VSSetConstantBuffers(
+                d3d11CommandBuffer->context,
+                d3d11UniformBuffer->bindSlots[i].slot,
+                1,
+                &nullBuf
+            );
 
-	SDL_UnlockMutex(renderer->acquireCommandBufferLock);
+            ID3D11DeviceContext1_VSSetConstantBuffers1(
+                d3d11CommandBuffer->context,
+                0,
+                1,
+                &d3d11UniformBuffer->bufferContainer->activeBuffer->handle,
+                &offsetInConstants,
+                &blockSizeInConstants
+            );
+        }
+        else if (d3d11UniformBuffer->bindSlots[i].shaderStageFlag == SDL_GPU_SHADERSTAGE_FRAGMENT)
+        {
+            /* stupid workaround for god awful D3D11 drivers
+             * see: https://learn.microsoft.com/en-us/windows/win32/api/d3d11_1/nf-d3d11_1-id3d11devicecontext1-vssetconstantbuffers1#calling-vssetconstantbuffers1-with-command-list-emulation
```

Comment:
maybe update this url to https://learn.microsoft.com/en-us/windows/win32/api/d3d11_1/nf-d3d11_1-id3d11devicecontext1-pssetconstantbuffers1#calling-pssetconstantbuffers1-with-command-list-emulation but i assume the meaning is the same